docs/src/userguide/wrapping_CPlusPlus.rst

```diff
@@ -440,6 +440,13 @@ called, which allows one to do custom C++ to Python error "translations." If
 raise_py_error does not actually raise an exception a RuntimeError will be
 raised.
 
+There is also the special form::
+
+    cdef int raise_py_or_cpp() except +*
```

Comment:
This suggests that `except +` is rather orthogonal to the Python exception return value. But `except +-1` looks funny … and it's very ambiguous.

Does `@cython.exceptval('+')` actually work in pure mode? There could be a way out in that corner …

Reply:
It is orthogonal, and there's not really a good way to mix them. The exception value is a return value in the normal case, and a ~error handler in the + case. One might want a return value or an error handler or neither or both. 

Right now, however, you can't safely use a function that may throw C++ or Python exceptions. 

I do like the idea of using decorators, perhaps with keyword arguments rather than magic characters.